tools/cli/adminDBCleanCommand.go

```diff
@@ -0,0 +1,309 @@
+		FailedCleanedDirectoryPath       string
+	}
+
+	// ShardCleanOutputFiles are the files produced for a clean of a single shard
+	ShardCleanOutputFiles struct {
+		ShardCleanReportFile    *os.File
+		SuccessfullyCleanedFile *os.File
+		FailedCleanedFile       *os.File
+	}
+)
+
+// AdminDBClean is the command to clean up executions
+func AdminDBClean(c *cli.Context) {
+	lowerShardBound := c.Int(FlagLowerShardBound)
+	upperShardBound := c.Int(FlagUpperShardBound)
+	numShards := upperShardBound - lowerShardBound
+	startingRPS := c.Int(FlagStartingRPS)
+	targetRPS := c.Int(FlagRPS)
+	scaleUpSeconds := c.Int(FlagRPSScaleUpSeconds)
+	scanWorkerCount := c.Int(FlagConcurrency)
+	scanReportRate := c.Int(FlagReportRate)
+	if numShards < scanWorkerCount {
+		scanWorkerCount = numShards
+	}
+	inputDirectory := getRequiredOption(c, FlagInputDirectory)
+
+	rateLimiter := getRateLimiter(startingRPS, targetRPS, scaleUpSeconds)
+	session := connectToCassandra(c)
+	defer session.Close()
+	cleanOutputDirectories := createCleanOutputDirectories()
+
+	reports := make(chan *ShardCleanReport)
+	for i := 0; i < scanWorkerCount; i++ {
+		go func(workerIdx int) {
+			for shardID := lowerShardBound; shardID < upperShardBound; shardID++ {
+				if shardID%scanWorkerCount == workerIdx {
+					reports <- cleanShard(
+						rateLimiter,
+						session,
+						cleanOutputDirectories,
+						inputDirectory,
+						shardID)
+				}
+			}
+		}(i)
+	}
+
+	startTime := time.Now()
+	progressReport := &CleanProgressReport{}
+	for i := 0; i < numShards; i++ {
+		report := <-reports
+		includeShardCleanInProgressReport(report, progressReport, startTime)
+		if i%scanReportRate == 0 {
+			reportBytes, err := json.MarshalIndent(*progressReport, "", "\t")
+			if err != nil {
+				ErrorAndExit("failed to print progress", err)
+			}
+			fmt.Println(string(reportBytes))
+		}
+	}
+}
+
+func cleanShard(
+	limiter *quotas.DynamicRateLimiter,
+	session *gocql.Session,
+	outputDirectories *CleanOutputDirectories,
+	inputDirectory string,
+	shardID int,
+) *ShardCleanReport {
+	outputFiles, closeFn := createShardCleanOutputFiles(shardID, outputDirectories)
+	report := &ShardCleanReport{
+		ShardID: shardID,
+	}
+	failedCleanWriter := NewAdminDBCorruptedExecutionBufferedWriter(outputFiles.FailedCleanedFile)
+	successfullyCleanWriter := NewAdminDBCorruptedExecutionBufferedWriter(outputFiles.SuccessfullyCleanedFile)
+	defer func() {
+		failedCleanWriter.Flush()
+		successfullyCleanWriter.Flush()
+		recordShardCleanReport(outputFiles.ShardCleanReportFile, report)
+		deleteEmptyFiles(outputFiles.ShardCleanReportFile, outputFiles.SuccessfullyCleanedFile, outputFiles.FailedCleanedFile)
+		closeFn()
+	}()
+	shardCorruptedFile, err := getShardCorruptedFile(inputDirectory, shardID)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			report.Failure = &ShardCleanReportFailure{
+				Note:    "failed to get corruption file",
+				Details: err.Error(),
+			}
+		}
+		return report
+	}
```

Comment:
Add `defer shardCorruptedFile.Close()` after this `if` statement

Reply:
Good catch thank you